model/src/main/kotlin/ScanRecord.kt

```diff
@@ -63,8 +70,21 @@ data class ScanRecord(
      */
     @Suppress("UNUSED") // Not used in code, but shall be serialized.
     val hasIssues by lazy {
-        scanResults.any { scanResultContainer ->
-            scanResultContainer.results.any { it.summary.issues.isNotEmpty() }
+        scanResults.any { (_, results) ->
+            results.any { it.summary.issues.isNotEmpty() }
+        }
+    }
+}
+
+private class ScanResultsDeserializer : StdDeserializer<SortedMap<Identifier, List<ScanResult>>>(
```

Comment:
Would it make sense to add a code comment saying that this is for backwards compatibility?

Reply:
Done. I have also added a commit to add the same comment for the `AdvisorResultsDeserializer`.